core/field.ts

```diff
@@ -1263,6 +1264,16 @@ export abstract class Field<T = any>
     return false;
   }
 
+  /**
+   * Get the variable model if this field refers to one.
+   *
+   * @returns The field's variable, or null if none exists.
+   * @internal
+   */
+  getVariable(): VariableModel | null {
```

Comment:
Can you also change this to return an array instead of a single model, and update the other logic accordingly? If we're fixing up this code anyway might as well  make it more general :P

Reply:
Sure, although `getVariable()` is already used in a few places, which might make it a breaking change too?

Reply:
I think I'm going to keep it singular in round one :smile: Perhaps we can do a follow up commit for the plural case when everything else is sorted and squash it.